### ISSUE TITLE
Release on Flathub

### ISSUE THREAD

**Issue by wvengen** [artisan-roaster-scope/artisan#1602]
Artisan is packaged for Linux as AppImage, Debian package and RPM. It's great to have this.
I would like to suggest to also offer a [Flatpak](https://flatpak.org/) package on [Flathub](https://flathub.org/), because:
- Many Linux distributions nowadays include this as a 'software store', providing more integrated discovery and installation. I think this would lower the barrier of installation for some people.
- Integrated updates (vs. downloading and installing a new version from Github Releases)
- The existing packages bundle most dependencies. Flatpak uses existing runtimes that can be shared with other applications, providing both download, on-disk and run-time memory savings.
- Possible downside: I'm not sure if the udev rules can be made to work automatically.

I would be willing to work on this, and have [a mostly working Flatpak already](https://github.com/wvengen/flathub/tree/artisan). Pending:
- [x] Come up with a cleaner way of installation (it's wasn't fully clear to me how to build this without pyinstaller bundling) - will tackle this later, after the first Flatpak release ([comments](https://github.com/artisan-roaster-scope/artisan/issues/1602#issuecomment-2241157072))
- [x] Make PDF export of reports work
- [x] App id - `org.artisan_scope.artisan` (dash becomes underscore, see [here](https://docs.flathub.org/docs/for-app-authors/requirements/#application-id))
- [x] Check if we can cleanup more (the app package is now 663MB ~~624MB~~ ~~539MB~~ unpacked; note that the deb is 727MB unpacked including Qt) - remove build dependencies; symlinked fonts already present in base system; prune yoctupuce binary libraries; remove QtWebEngine devtools, webdriver and dictionaries; ~~also removed babel and QtWebEngine locales for languages which Artisan is not translated into~~
- [x] ~~Optionally split locales to separate packages~~ doesn't do much yet for this app, skip for now
- [x] Verify it works with attached roasters
- [x] Fix graph not updating in real-time (env var / debug matplotlib) -> fixed in v2.10.5-beta
- [x] Complete any missing metadata (metainfo, desktop file, icon, screenshot)
- [x] Review `finish-args` (do we need sounds? x11-fallback? ...)
- [x] Make switching language work
- [x] Make secret storage work
- [x] Update to Artisan v3 (when released)
- [x] Check last-saved / last-opened location approach
- [x] Submit
- [ ] Get app verified in Flathub

**Comment by MAKOMO** [comment]
Sure. Contributions like this are very welcome. However, we would need to integrate this into our current CI process to have those flatpacks generated automatically.

Note that creating installers for Artisan is not trivial. For example, Artisan uses Qt's QWebEngine to render the HTML reports internally and generate PDF exports. Correctly bundling QWebEngine is non-trivial.

We wish the builds would be smaller (the universal2 builds for macOS are larger than 1.2GB)! However, there are limits. Artisan uses almost anything from Qt. Not much to keep off there (the macOS build script does a lot removal and still the packages are large). 

Also the translations/locales add considerable to the size of the packages. Not sure if it makes sense to split them off. How would the correct locale be installed per user? Besides that some users use several locales at the same time (I have my locale in english, but use software often set to German).

Regarding the app id: the github home of Artisan has a dash in the name as well. Maybe you can/have-to escape those dashes?

**Comment by wvengen** [comment]
Thank you for your response, and happy to hear you're open to this.

Note that the building happens on Flathub infrastructure, ideally based on source code for all components. CI could be useful to ensure it builds correctly, but is not strictly necessary. There is a KDE+Qt + PyQt basis, I will have a look how QtWebEngine integrates into this.
I think it would be helpful to have a way to install the application, without bundling it (i.e. letting the system integrator handle dependencies). Something a bit cleaner than [what I have now](https://github.com/wvengen/flathub/blob/2833b25b93487d83f29b73b01b76174bf84b5e2c/org.artisanscope.artisan.yml#L34-L49). This could also be a shellscript, a `setup.py`, or whatever. Possibly related: #831.

Translations/locales can either be bundled in the Flatpak, or be present as a kind of language-add-on packages (there is infrastructure for this). There can be multiple language packages installed, so that should be fine.

I'll ask Flathub for direction on what to do with the dashes.

**Comment by wvengen** [comment]
I think the Flatpak packaging is almost finished.
There are some things I need your help with, before I can [submit this](https://docs.flathub.org/docs/for-app-authors/submission) to Flathub.

## Appstream file

Inclusion of an [appstream](https://en.wikipedia.org/wiki/AppStream) metadata file in the artisan repository (see Flathub on [Required Metadata](https://docs.flathub.org/docs/for-app-authors/requirements/#required-metadata)). This file is used to show the app in appstores, across different distributions. It contains description, links to screenshots, relevant URLs, and information about releases. Note that this file needs to be updated with each release (or the new release cannot be released on Flathub). If this is tedious, I can help to think about how to automate updating this, e.g. from Github releases and the release history.

### Also for Debian & other Linux distributions

Note that the appstream file may also be useful to include in other Linux packages. For a Debian package, see [Debian AppStream Guidelines](https://wiki.debian.org/AppStream/Guidelines). Note that the `id` is meant to be equal across distributions (used e.g. when Artisan is available as a system package and a Flatpak, the software store will show this as one entry, allowing a user to pick which version to install). So this file could be installed as `/usr/share/metainfo/org.artisan_scope.artisan.metainfo.xml`. This also means the desktop file in `/usr/share/applications` should be renamed to `org.artisan_scope.artisan.desktop`.

Perhaps this is something to consider later. It is mostly relevant when the deb package is included in a repository that software stores can recognize. I would just like you to know about this, to decide where to put the appstream file.

### Open question on OARS

[OARS](https://hughsie.github.io/oars/) is about age rating metadata, and there is one question where I did not have the answer for: is any user tracking happening (e.g. a user counter, error reporting, or something similar)? If no, then the `social-info` line can be removed.

Note that the `money-purchasing` line is present because donations are requested (and artisan.plus integration may also count).

### The Appstream file

<details>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<component type="desktop-application">
  <id>org.artisan_scope.artisan</id>
  
  <name>Artisan</name>
  <summary>Visual scope for coffee roasters</summary>
  
  <metadata_license>CC0-1.0</metadata_license>
  <project_license>GPL-3.0-only</project_license>
  
  <supports>
    <control>pointing</control>
    <control>keyboard</control>
    <control>touch</control>
  </supports>

  <description>
    <p>Artisan is an open-source software that helps coffee roasters record, analyze, and control roast profiles. This software can automate the creation of roasting metrics to help make decisions that influence the final coffee flavor.</p>

    <p>Notable features</p>
    <ul>
      <li>we believe in Open-Source, check out our code and contribute</li>
      <li>seamless integration into the artisan.plus inventory management service</li>
      <li>extensive machine and device support including control</li>
      <li>unlimited number of curves</li>
      <li>rate-of-rise (RoR), area-under-the-curve (AUC), development-time-ratio (DTR) calculations and projection lines</li>
      <li>statistics, roast profile evaluation, Profile Analyzerg, Roast Comparatorg, Profile Transposerg and Roast Simulatorg</li>
      <li>roast-, production- and ranking reportsg</li>
      <li>automated reproduction of roasts via alarm programsg, replay of eventsg or PID controlg</li>
      <li>batch counterg</li>
      <li>profile designerg, cupping editor, spider- and wheel graph</li>
      <li>user defined buttons and slidersg with programmable actions</li>
      <li>many import and export formats like Aillio Roastime, Cropster XLS, Giesen CSV, IKAWA CSV, Probat Pilot, RoastLogger, RoastLog, RoastPath,..</li>
    </ul>

    <p>The Artisan project runs on donations from individuals and companies recognising the value of Artisan for their work or leisure. Consider supporting this project with your donation, or even better, purchase an artisan.plus membership!</p>
  </description>

  <developer id="org.artisan-scope">
    <name>Artisan developers</name>
  </developer>

  <url type="bugtracker">https://github.com/artisan-roaster-scope/artisan/issues</url>
  <url type="homepage">https://artisan-scope.org/</url>
  <url type="donation">https://artisan-scope.org/donate/</url>
  <url type="contact">https://artisan-roasterscope.blogspot.com/p/contact-me.html</url>
  <url type="faq">https://artisan-scope.org/help/#faq</url>
  <url type="vcs-browser">https://github.com/artisan-roaster-scope/artisan</url>
    
  <launchable type="desktop-id">org.artisan_scope.artisan.desktop</launchable>
  <screenshots>
    <screenshot type="default">
      <image>https://github.com/artisan-roaster-scope/artisan/blob/2c94e7d03f51221771a6f5e10c3a55fd1a26e744/wiki/screenshots/artisan-cover.png?raw=true</image>
    </screenshot>
    <screenshot>
      <image>https://github.com/artisan-roaster-scope/artisan/blob/2c94e7d03f51221771a6f5e10c3a55fd1a26e744/wiki/screenshots/artisan-cover2.png?raw=true</image>
    </screenshot>
  </screenshots>

  <content_rating type="oars-1.1">
    <content_attribute id="social-info">mild</content_attribute> <!-- TODO check if there is a user-counter or so (e.g. with update check) -->
    <content_attribute id="money-purchasing">mild</content_attribute>
  </content_rating>

  <branding>
    <color type="primary" scheme_preference="light">#d8ebf1</color>
    <color type="primary" scheme_preference="dark">#0c6aa6</color>
  </branding>

  <releases>
    <release version="2.10.4" date="2024-03-22">
      <url type="details">https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.10.4</url>
      <description>
        <p>This version is a bug fix release.</p>

	<ul>
          <li>adds metadata to PDF, SVG and PNG exports</li>
	  <li>fixes regression which broke the PID dialog for MODBUS, S7, TC4 and Kaleido (Issue #1480 and Issue #1515)</li>
	  <li>fixes regression which broke devices Omega HH309 34 and the just introduced Digi-Sense 20250-07 and Extech 42570 (PR #1481)</li>
	  <li>fixes regression introduced in v2.8.4 which broke persisting volume changes made in Roast Properties dialog (Discussion #1511)</li>
	  <li>fixes regression which broke persistence of custom column width of alarm and extra device and table</li>
	  <li>fixes regression which broke the formating of extra curve data in background event table</li>
	  <li>fixes regression which broke the automatic opening of completed profiles in ArtisanViewer (Issue #1516)</li>
	  <li>improves formating of profile and background data tables, custom event button table, alarm table and extra device table</li>
	  <li>fixes Linux builds that failed to start Artisan v2.10.2 on some configurations</li>
	  <li>fixes phases bar rendering in case CHARGE is not set</li>
	</ul>

	<p>IMPORTANT NOTE</p>
	<p>The following issues have found in this release and are being worked on:</p>
	<ul>
          <li>this version does not play well with Probat Series III and Sample Roaster on all supported platforms. Please keep using the v2.8.4 release until v2.10.6 has been released</li>
	</ul>
      </description>
    </release>
  </releases>
</component>
```

</details>

## Cleaning up the installation script

The Flatpak currently contains some lines to install the package. It does not use pyinstaller, also because some dependencies are already installed in the runtime and base application (it wouldn't make sense to bundle those).

[This is](https://github.com/wvengen/flathub/blob/fef09d292fb3887b40f58a7b487564c524098559/org.artisan_scope.artisan.yml#L59-L79) what I currently do. I would love to have most of these steps as an installer script in the Artisan source tree: something that installs the app in a directory from which it is runnable, without bundling or dependencies.

One thing I encountered is a Python recursion error on import of `snap7`. Moving this import to earlier in the process made this error disappear, that is the `sed` line on `artisan.py` for. Could not pinpoint the underlying cause though.

I could try my hand at that, but would need to refactor the build scripts. Please let me know if you think that's a good idea.

## Icon

Is the icon available as SVG? That is preferred nowadays.

**Comment by MAKOMO** [comment]
I added tyour appstream metadata file under debian/usr/share/metainfo and the app icon as artisan.svg. I also renamed the desktop file according to your instructions. Regarding OARS, the information on the page you link is a little loose. The Artisan app is counting the number of roasts a user does to show a dialog asking for a donation on a regular basis. However, this information is not transmitted anywhere and stays local. Then there is a

**Comment by MAKOMO** [comment]
Could those build scripts for flatpacks/flathub also be independent of what we have right now?

One more thing: we are about to release Artisan v3 (new major version) next week or latest the week after. This could be the first release on Flathub. We could go for a "manual" flatpack build for this one and it could also be submitted a little after the regular v3 release. 

Let us know what is possible!

**Comment by wvengen** [comment]
Thanks for your replies, and for adding the appstream metadata file, and icon (the icon still contains embedded PNGs for something, I'll have a look if I can make it fully SVG-based).

Regarding OARS, I would judge the optional debugging info sent manually by mail not as user tracking, good you removed the line.

I'll have a look at the build scripts. In the meantime, I feel confident enough to submit this in its current form to Flathub, then it can hopefully be simplified later.

The snap7 thing appeared to be caused by the [importlib monkey patch for MacOS](https://github.com/artisan-roaster-scope/artisan/blob/cc0eb9e5a4e1d52e123525a64dd6817bc9c31abc/src/artisanlib/main.py#L79-L91), that recently seems to have been disabled (but not in the latest release 2.10.4, which I'm working with).

The build scripts for the Flatpak can indeed be independent of the rest. But I do see quite some duplication across the build scripts, and I would think it is hard to maintain a growing number of them, especially e.g. what is copied to `dist/`. I think it could help other packagers to have a script that installs all relevant files to run from a certain location, without dependencies. (And even pyinstaller could start from that, to keep the process clear across platforms.) I'll see what I can do in a PR.

Great to hear about v3, I'll be happy to get that submitted to Flathub. Agree that we can keep the Flathub building aproach as it is now, and gradually improve (by moving some parts to build scripts part of artisan), if that turns out to make sense.

**Comment by MAKOMO** [comment]
I will update the SVG with a clean one in my next commit. Sorry.

Right, this monkey patch was necessary as packaging on macOS failed. At least this one is resolved now. We constantly have to work around those things. Currently the main Python packaging lib introduced several complex incompatibilities  one version just with py2app another one with pip/pyinstaller/numpy/scipy. Hard to find a combination of versions per platform which allows to bundle builds together:(

**Comment by wvengen** [comment]
Thank you! No problem :)

Ah, that's why there are all these version number specifics in `requirements.txt`. I imagine the number of files explodes when not bundling, with all those dependencies. Thanks for all the hard work!

**Comment by wvengen** [comment]
I'm trying out v2.10.5-beta now, and notice that `pydantic` is now a required dependency, also for running. Could we make this optional? It is a dependency that brings in no run-time functionality (as far as I understand), and it requires rust to compile from source, making the build more complex. "No" is also ok, then I'll start integrating rust.

**Comment by wvengen** [comment]
Regarding pydantic: I have it working now. The total Flatpak size is 564 MB without and 585 MB with pydantic (+additional dependencies).

**Comment by MAKOMO** [comment]
Pydantic is needed and we plan to make use of pydantic within Artisan in more places in the future. Thanks for your work thus I don't understand why those python packages cannot be just downloaded from pypip via pip as the Artisan build scripts do. In that case there seems to be no need for use rust to compile from source, right?

**Comment by wvengen** [comment]
Thanks for your reply! Good to hear more use of pydantic is planned (for the code), and to know what to aim for. My reasoning was, that often, checks done on development are omitted in production builds, I would have expected pydantic to allow this, some kind of 'dummy' mode where these checks are omitted (unless we _want_ more checks in production builds). But guess that would be more something for upstream.

The Flatpak packaging scripts prefer building things from source, I think both because having the source is generally more flexible / debuggable / reproducible, but also just because that's what the packaging scripts do right now (it is not enforced, however).
One important trait of the Flatpak approach is that no network access is permitted during build, so all possibly used dependencies (for all architectures) need to be declared beforehand.

In any case, the issue is tackled.

**Comment by MAKOMO** [comment]
We use (and plan to use more) pydantic to validate input on read into the app during runtime. Those check we want during runtime not to have to handle such "broken" data within the app.

For development checks we use the type checkers mypy and pyright as well as the linters pylint and ruff. Those type annotations used by the type checkers are indeed ignore by the Python runtime.

I understand this idea of working from source. However, many packages Artisan is build upon are non-trivial to build from source correctly, like PyQt, numpy, scipy and some others. The chance is high that, without that expert knowledge of those package owners, your builds are somewhat broken, at least on some platforms. Therefore other Python packages and apps like Artisan using those packages load the corresponding binary builds from pypip. Those packages can be retrieved before the "build" (assemble) on Flatpack to avoid any further network communication during the build. This is how the CI Artisan builds work too.

**Comment by wvengen** [comment]
Thanks for the background, good to know!

The idea that building from source might lead to a broken package is a something I need to think about. PyQt has a Flatpak BaseApp to work with, so should be fine. Numpy, scipy and matplotlib, as well as dependencies like cryptography are widely built from source in Flatpaks. There are some packages that need native libraries, which I have added where known (when not shipped in the Python package). I'll take a critical look at them again.

**Comment by MAKOMO** [comment]
Isn't it very inefficiency (energy wise) if each install compiles every component from source, besides that the install takes way longer?

**Comment by wvengen** [comment]
Ah, that would be cumbersome and wasteful, yes. What is installed by the user is in binary form (using [OSTree](https://netmodule-linux.readthedocs.io/en/latest/gettingstarted/3_ostree.html)) (served by Flathub), the build process builds from source.

**Comment by wvengen** [comment]
Question on secret storage (for the plus integration, afaik): [keyring](https://github.com/jaraco/keyring) comes with both `SecretStorage` and `libsecret` backends out of the box (the latter is required for Flatpak, for the sandboxing of secret access to work properly). Artisan picks up a specific keyring backend in the code, is there a reason not to use keyring's [default backend selection](https://github.com/jaraco/keyring?tab=readme-ov-file#configuring)? That would make libsecret work automatically, and could clean up the Artisan code related to setting up the keyring (also giving users an option to override it in the environment, if needed).

**Comment by wvengen** [comment]
Ouch, that sounds like a bad situation. Good to hear it's working now, and yes, a little scray ... I will test on my Linux setup before the next release.

Even though documentation suggests that libsecret works with Flatpak for the sandboxed environment, I could not get it to work without adding permissions to access all secrets (it so happens that all applications I could find that work with passwords do this as well). And then the existing SecretStorage also works (the Python keyring packages actually prefers it). So no change needed here (it will probably at some point in the future, when sandboxing is improved).

I'll try to keep an eye on this in the future.

**Comment by MAKOMO** [comment]
v3 released. What is the status on Flathub?

**Comment by wvengen** [comment]
Thanks for the headsup! I'm on it, will do some testing, and create the PR on Flathub. Since a lot happens to actually install Artisan, I expect some questions, but think that in a week or so it could be available (depending a bit on how much work still is needed, according to the Flathub reviewer).

**Comment by wvengen** [comment]
Ah, one thing: I'm missing the release info in the metainfo file. This is how the Flatpak finds out the version, so it is pretty important to add the release to this file for each release (sorted by release date, newest on top). Could you still add that? (Strictly speaking, this may be a new release, but I can also get the updated file from Github during the build, without doing a new release.)

**Comment by wvengen** [comment]
Ah, I found a way to do this in the Flatpak repository, so no hurry I guess. You could add this diff, if you like.
```
diff --git a/src/debian/usr/share/metainfo/org.artisan_scope.artisan.metainfo.xml b/src/debian/usr/share/metainfo/org.artisan_scope.artisan.metainfo.xml
index 8303a20c..7950ddc5 100644
--- a/src/debian/usr/share/metainfo/org.artisan_scope.artisan.metainfo.xml
+++ b/src/debian/usr/share/metainfo/org.artisan_scope.artisan.metainfo.xml
@@ -67,6 +67,52 @@
   </branding>
 
   <releases>
+    <release version="3.0.0" date="2024-09-01">
+      <url type="details">https://github.com/artisan-roaster-scope/artisan/releases/tag/v3.0.0</url>
+      <description>
+        <p>ADDITIONS</p>
+	<ul>
+          <li>adds support for the execution of roast plans scheduled on artisan.plus</li>
+          <li>adds advanced summary statistics</li>
+          <li>adds support for transparent colors</li>
+          <li>adds {WEIGHTin} placeholder substitute by the current batch size (g) in command actions</li>
+          <li>adds additional button label substitutions, \V, \F and \T reporting the event value, the event value interpreted as temperature in Fahrenheit, and the event value interpreted as temperature in Celsius. The last two are automatically converted to the currently selected temperature unit.</li>
+          <li>adds "Load p-i-d from background" setting to configure the PID to the settings stored in background profile</li>
+          <li>capture and persist between-batch protocol (BBP) data as measured during a roasting session</li>
+	</ul>
+
+	<p>NEW HARDWARE SUPPORT</p>
+	<ul>
+          <li>adds machine support for the BeanGo Cube X</li>
+          <li>adds machine support for MUGMA Roasters</li>
+          <li>adds updated Sivetz fluid bed roasting machines support for the latest machines recording also fan changes</li>
+          <li>adds machine support for iRm roasting machines featuring Mitshubishi PLCs</li>
+          <li>adds Bühler RM20 Simatic Legacy setup supporting older firmware versions not returning the machine state (Issue #1529)</li>
+          <li>adds support for the Phidget RCC0004 server motor controller (Discussion #1546)</li>
+          <li>adds support for the to-be-release Phidget TMP1202 module</li>
+          <li>adds Hottop Command control to activate and deactivate the control function of the machine via event buttons and alarms</li>
+          <li>adds compression toggle, detailed device logging, and origin header to WebSocket communication</li>
+	</ul>
+
+	<p>CHANGES</p>
+	<ul>
+          <li>only reset roasting notes on RESET if profile is loaded (Issue #1521)</li>
+          <li>disable input filtering on device channel of binary or special types like NONE, dummy, and slider values</li>
+        </ul>
+
+	<p>FIXES</p>
+	<ul>
+          <li>fixes regression preventing Artisan v2.10.2 and v2.10.4 to connect successfully to Probat Series III machines via WebSockets incl. the Probat Sample Roaster (Issue #1531)</li>
+          <li>fixes regression in Roast Properties, unable to change/add ground color value (Issue #1520)</li>
+          <li>fixes regressions in table copy functions</li>
+          <li>fixes regression introduced in v2.8.4 that prevented the cupping chart being added properly to roast reports (Discussion #1563)</li>
+          <li>fixes event playback by temperature being blocked by already past background events if playback is turned ON during a roast</li>
+          <li>avoids rendering timestamps as "xx:60" in mouse pointer time/temp/RoR widget</li>
+          <li>ensures that time-axis ticks extend over the full range of readings w.r.t. the loaded background and foreground profiles</li>
+          <li>fixes a regression preventing the correct persistence of default SV values (Issue #1631)</li>
+        </ul>
+      </description>
+    </release>
     <release version="2.10.4" date="2024-03-22">
       <url type="details">https://github.com/artisan-roaster-scope/artisan/releases/tag/v2.10.4</url>
       <description>
```

**Comment by MAKOMO** [comment]
Sorry for being late, I forgot to update that metainfo.xml. Did it now and corrected the release date (1.9 => 1.8).

**Comment by wvengen** [comment]
Thanks for updating the metainfo file.
I think the Flathub PR is ready to be merged, but is taking some time (perhaps because of the holiday season?). I'll refrain from making changes to the PR until merged (because it might need another review round if I do), and then I'll update the release date.

**Comment by MAKOMO** [comment]
Something just happend. I got an invite to join flathub/org.artisan_scope.artisan and did so. Anything further I need to do?

**Comment by wvengen** [comment]
The app [is live on Flathub now](https://flathub.org/apps/org.artisan_scope.artisan)!
Correcting the release date in https://github.com/flathub/org.artisan_scope.artisan/pull/1. I'll merge that to correct the release date on Flathub.

Next step would be to [get the app verified](https://docs.flathub.org/docs/for-app-authors/verification/).

**Comment by MAKOMO** [comment]
Working on verification...

**Comment by MAKOMO** [comment]
I did add the verification token to the artisan-scope.org site and clicked "Verify" under the Flathub Developer Portal. It now shows "Unverify", but the Artisan entry on Flathub still shows Unverified. Maybe this needs a moment? There is also a node on the Artisan Flathub page which states "App data review pending". Some entries are ticked already, most are pending. Anything we have to do on this?

**Comment by wvengen** [comment]
And verified for me.
![image](https://github.com/user-attachments/assets/c4bd7d96-b4e0-48a9-b8fe-32a5c810a01d)

**Comment by MAKOMO** [comment]
Cool! What a success. The linux download page already links to Flatpack.

Thanks a lot for all that time you spent on this! Wonderful contribution! We couldn't have done this on our own as we lack the required compentences. Hope you are also a user and maybe you'd like to join the development team or at least the Linux-dev/release team;)

Feel free to close this issue if you think this is done.

**Comment by wvengen** [comment]
Thank you for your kind words. I'm a happy (though fairly fresh) user, and would be glad to be involved in Linux-dev/release areas of Artisan.

I think my next step would be looking at the installation and bundling process for the various platforms, for which I'll open a new issue.

**Comment by MAKOMO** [comment]
We are releasing Artisan v3.0.2, a patch release, today. I did update the metainfo.xml in the repository and also added a [screenshot of Artisan 3 running on linux](https://github.com/artisan-roaster-scope/artisan/blob/master/wiki/screenshots/artisan-cover-linux.png) to the repository.

Just received an email from the Flathub.Bot that the build fails. Not sure what triggered this build. Anyhow, I could not even figure out the reason why and what exactly failed from the provided log. Could you take a look?

Thanks!

**Comment by wvengen** [comment]
Thank you for the new release, and adding a Linux screenshot.
The Flathub package periodically checks if a new release has been made (with the external data checker information specified).
This auto-created a PR in https://github.com/flathub/org.artisan_scope.artisan/pull/2
But we need more complex dependency handling, so auto-created PRs don't work here. I'll have a look at it this week.

**Comment by MAKOMO** [comment]
Thanks for this information and taking care. There is no hurry on this, this release just fixes a build issue that breaks the new (non-essential) scheduler feature, makes some smaller adjustments and updates some libs. We could have gone with just a build update on Linux and Windows, but we decided that the performance increase of using the ruff-based bidi implementation and that of the new numpy/scipy major versions would be worse a minor release.

**Comment by wvengen** [comment]
Seems to run fine. And nice to see the numpy major version bump (I'll do some more math-related testing then).
The about screen is different, not sure why. Will look into it later.

![image](https://github.com/user-attachments/assets/7f1b842e-d094-40f4-aa12-b6591e0c8b37)

![image](https://github.com/user-attachments/assets/55b1a613-7eb6-4e76-b03b-df11762efe17)

**Comment by MAKOMO** [comment]
Excellent news! I removed the list of Contributors from the About file as this one was not maintained and to simplify things. Financial contributions are listed in the corresponding release blog posts and code contributions are managed by GitHub.

**Comment by wvengen** [comment]
Numpy and scipy tests were fine, merged the Flathub PR.
Would be happy to join the projects dev team. Thank you!

**Comment by MAKOMO** [comment]
We are about to release Artisan v3.1.2 those days. Anything specific we have to prepare for the Flathub release or should this work out automatically? No big changes w.r.t. libs or additional resources, just minor version updates of libs and source updates. Let me know!

**Comment by wvengen** [comment]
Thanks for the headsup! Current master (at https://github.com/artisan-roaster-scope/artisan/commit/74aa7dc657639984cc4a097b4bc0d3c19393af64) seems to be building and running well as a Flatpak.

**Comment by MAKOMO** [comment]
Excellent! Then just go forward with this release. I am preparing all the other builds right now and will upload them to GitHub over the day. I will also prepare announcements and others.

Thanks for your support on this!
M

**Comment by wvengen** [comment]
Thanks :) I'm not fully online this week, but will try to update Flathub. I would need a tag to publish, so I'll wait for that. Other than that, I'm all set. And great to see another release :)

**Comment by MAKOMO** [comment]
Ok, the release tag is here now: https://github.com/artisan-roaster-scope/artisan/releases/tag/v3.1.2

Take your time, I am still signing, notarizing, documenting,.....

**Comment by wvengen** [comment]
Released 1.3.2 on Flathub. Needed to simplify the metainfo file (nested lists are not supported), see [here](https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines#description), commited here in 5793be98e1dabd9155e5b2be7a8a850881969328.

**Comment by MAKOMO** [comment]
Thanks for your work on this. Sadly we discovered an issue in this release and are just discussing to fix this and do a quick re-release before we get too many complaints. I will keep you posted here.

**Comment by wvengen** [comment]
Thanks for mentioning. No problem, please let me know (here) when there is a new release, and I'll publish it to Flathub when I can.

**Comment by MAKOMO** [comment]
We just released v3.1.4 (hopefully) improving the backward compatibility with the problematic MODBUS setups. Only version number, a few text files and one source file changed. I updated the changelog files accordingly. If you find time to do another release on flathub....

Thanks for your support!
M

**Comment by MAKOMO** [comment]
Just released Artisan v3.2 and Flathub failed on org.artisan_scope.artisan.metainfo.xml. Hm. I did check this file with an XML validator and it looked good. Can you see the issue?

**Comment by wvengen** [comment]
Last release I needed a patch to the metainfo file, but this is not needed anymore now - I will remove it.
Thanks for checking :) I'll pick up the release this week (maybe today), unfortunately, an update needs some custom scripts to handle the dependencies, so it doesn't go automatically yet.

**Comment by MAKOMO** [comment]
Ok! Thanks for taking this. No hurry!

**Comment by MAKOMO** [comment]
We did just release a new version of Artisan. Artisan v3.4. Hopefully this time all is in good shape such that there are now hurdles to create a new flathub release. There are no major differences in terms of build process or dependencies. Let's see how this goes.

**Comment by wvengen** [comment]
Thank you for the note! That was indeed pretty straightforward, well done!
https://github.com/flathub/org.artisan_scope.artisan/pull/29

**Comment by MAKOMO** [comment]
Wonderful! Thanks for your excellent contribution!
